spec/ffmpeg_spec.rb

```diff
@@ -89,21 +76,57 @@
         end
 
         it 'raises IO::TimeoutError' do
-          expect { described_class.ffmpeg_execute(*args) }.to raise_error(IO::TimeoutError)
+          expect { described_class.ffmpeg_execute!('hello', 'world') }.to raise_error(IO::TimeoutError)
         end
       end
 
       context 'with operation timeout set' do
         it 'raises Timeout::Error' do
-          expect { described_class.ffmpeg_execute(*args, timeout: 0.5) }.to raise_error(Timeout::Error)
+          expect { described_class.ffmpeg_execute!('hello', 'world', timeout: 0.5) }.to raise_error(Timeout::Error)
         end
       end
     end
   end
 
   describe '.ffmpeg_execute!' do
     it 'raises an error when the process is unsuccessful' do
-      expect { FFMPEG.ffmpeg_execute!('-v') }.to raise_error(FFMPEG::Error)
+      expect { described_class.ffmpeg_execute!('-v') }.to raise_error(FFMPEG::Error)
+    end
+
+    context 'when called in a subprocess' do
+      before do
+        described_class.ffmpeg_binary = fixture_file('bin/mock-ffmpeg')
+      end
+
+      context 'with exit signal traps' do
+        it 'does not raise an error' do
+          pid = fork do
+            Signal.trap('QUIT') {} # rubocop:disable Lint/EmptyBlock
+
+            described_class.ffmpeg_execute!('-n=2', '-progress', 'hello', 'world')
```

Comment:
can we change it to `-n=3` to make the test more reliable?

Reply:
sure, just wanted to make the test execute quicker, because in this case we need to wait for it to finish